packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -882,6 +884,14 @@ export class GarbageCollector implements IGarbageCollector {
 
         // Update the current state and update the runtime of all routes or ids that used as per the GC run.
         this.updateCurrentState(gcData, gcResult, currentReferenceTimestampMs);
+        const referencedDataStoreNodeIds: string[] = [];
+        for (const key of gcResult.referencedNodeIds) {
+            if (this.runtime.getNodeType(key) === GCNodeType.DataStore) {
+                referencedDataStoreNodeIds.push(key);
+            }
+        }
+        this.runtime.revive(referencedDataStoreNodeIds);
+
         this.runtime.updateUsedRoutes(gcResult.referencedNodeIds);
```

Comment:
Maybe instead of adding `revive`, just update this guy to do it?

Reply:
I initially thought that, https://github.com/microsoft/FluidFramework/pull/12750#discussion_r1014118105, Navin had a different opinion on it.